### PR TITLE
Final Input Facade and Audio facade

### PR DESCRIPTION
**Pull request by RickvanBerlo**
This Pull Request (PR) adds functionality to the game according to User Story #XXX

This PR...

* [x] has added functionality.
* [x] complies with the code style of the repository.
* [ ] adds new unit tests where necessary.
* [x] is up to date with the `dev` branch.

Please merge with care!

### CODE REVIEW

File: memecity.engine/AudioFacade.cpp [hunk 1]
```diff
@@ -1 +1,88 @@
...
+	Mix_HaltChannel(channel);
+}
+
+Mix_Music* bgm = nullptr; //greater then 10 sec
+std::map<const char*, Mix_Chunk> chunks; // a list for all the soundtracks smaller then 10 sec 
+std::map<const char*, Mix_Chunk>::iterator it;
+
+void AudioFacade::Init() {
+	if (SDL_Init(SDL_INIT_AUDIO) < 0) {
+		std::cout << "video Initialization error: " << SDL_GetError() << std::endl;
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: Video?

File: memecity.engine/AudioFacade.cpp [hunk 2]
```diff
@@ -1 +1,88 @@
...
+	}
+}
+
+void AudioFacade::OpenAudio(int frequency, int channel, int chunksize) {
+	if (Mix_OpenAudio(frequency, MIX_DEFAULT_FORMAT, channel, chunksize) < 0) {
+		std::cout << "Audio error: " << Mix_GetError() << std::endl;
+	}
+}
+int AudioFacade::PlaySound(const char* name, int repeats, int volume) {
+	Mix_Chunk *sound = nullptr; // smaller then 10 sec
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: *than

File: memecity.engine/AudioFacade.cpp [hunk 3]
```diff
@@ -1 +1,88 @@
 ﻿#include "AudioFacade.h"
+
+void Stopchunck(int channel) {
+	Mix_HaltChannel(channel);
+}
+
+Mix_Music* bgm = nullptr; //greater then 10 sec
+std::map<const char*, Mix_Chunk> chunks; // a list for all the soundtracks smaller then 10 sec
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: Hoort dit niet in de `AssetManager` thuis?

File: memecity.engine/AudioFacade.cpp [hunk 4]
```diff
@@ -1 +1,88 @@
...
+
+	Mix_VolumeMusic(volume);
+
+	if (!Mix_PlayingMusic()) {
+		Mix_PlayMusic(bgm, -1);
+	}
+
+}
+
+void AudioFacade::PauzeBackGroundSound() {
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: `PauseBackgroundSound`

File: memecity.engine/AudioFacade.cpp [hunk 5]
```diff
@@ -1 +1,88 @@
...
+
+	Mix_VolumeChunk(sound, volume);
+	int channel = Mix_PlayChannel(-1, sound, repeats);
+	chunks.insert(std::pair<const char*, Mix_Chunk>(name, *sound));
+	Mix_ChannelFinished(Stopchunck);
+	sound = nullptr;
+	return channel;
+}
+
+void AudioFacade::PlayBackGroundSound(const char* name, int volume) {
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: `PlayBackgroundSound`

File: memecity.engine/AudioFacade.cpp [hunk 6]
```diff
@@ -1 +1,88 @@
...
+	{ //Resume the music 
+		Mix_ResumeMusic();
+	} //If the music is playing 
+	else
+	{ //Pause the music 
+		Mix_PauseMusic();
+	}
+}
+
+void AudioFacade::StopBackGroundSound() {
```
Comment (tvbronswijk) [inline memecity.engine/AudioFacade.cpp]: `StopBackgroundSound`

File: memecity.engine/InputFacade.cpp [hunk 7]
```diff
@@ -1 +1,25 @@
 ﻿#include "InputFacade.h"
+
+void InputFacade::update()
+{
+	while (SDL_PollEvent(&event) != 0)
+	{
+		if (event.type == SDL_QUIT) {} //TODO place Kill switch.
```
Comment (tvbronswijk) [inline memecity.engine/InputFacade.cpp]: Waarom is deze TODO nog niet afgehandeld?

**Review by MartijnPol — APPROVED** [review]
Looks fine, for the next time: please mind the feature branches!

**Pull request merged**